website/pages/docs/upgrading/instructions/general-process.mdx

```diff
@@ -0,0 +1,175 @@
+for the version you are on.
+
+## Download the New Version
+
+First, download the binary for the new version you want.
+
+<Tabs>
+<Tab heading="Binary">
+
+All current and past versions of the OSS and Enterprise releases are
+available here:
+
+- https://releases.hashicorp.com/consul
+
+</Tab>
+<Tab heading="Docker">
+
+Docker containers are available at these locations:
+
+- **OSS:** https://hub.docker.com/_/consul
+- **Enterprise:** https://hub.docker.com/r/hashicorp/consul-enterprise
+
+</Tab>
+<Tab heading="Kubernetes">
+
+If you are using Kubernetes, then please review our documentation for
+[Upgrading Consul on Kubernetes](/docs/k8s/operations/upgrading).
+
+</Tab>
+</Tabs>
+
+## Prepare for the Upgrade
+
+**1.** Take a snapshot:
+
+```
+consul snapshot save backup.snap
+```
+
+You can inspect the snapshot to ensure if was successful with:
+
+```
+consul snapshot inspect backup.snap
+```
+
+Example output:
+
+```
+ID           2-1182-1542056499724
+Size         4115
+Index        1182
+Term         2
+Version      1
+```
+
+This will ensure you have a safe fallback option in case something goes wrong. Store
+this snapshot somewhere safe. More documentation on snapshot usage is available here:
+
+- https://www.consul.io/docs/commands/snapshot
+- https://learn.hashicorp.com/tutorials/consul/backup-and-restore
+
+**2.** Temporarily modify your Consul configuration so that its [log_level](/docs/agent/options.html#_log_level)
+is set to `debug`. After doing this, issue the `consul reload` command on your servers. This will
+give you more information to work with in the event something goes wrong.
+
+## Perform the Upgrade
+
+**1.** Issue the following command to discover which server is currently the leader:
+
+```
+consul operator raft list-peers
+```
+
+You should receive output similar to this (exact formatting and content may differ based on version):
+
+```
+Node       ID                                    Address         State     Voter  RaftProtocol
+dc1-node1  ae15858f-7f5f-4dcb-b7d5-710fdcdd2745  10.11.0.2:8300  leader    true   3
+dc1-node2  20e6be1b-f1cb-4aab-929f-f7d2d43d9a96  10.11.0.3:8300  follower  true   3
+dc1-node3  658c343b-8769-431f-a71a-236f9dbb17b3  10.11.0.4:8300  follower  true   3
+```
+
+Take note of which agent is the leader.
+
+**2.** Copy the new `consul` binary onto your servers and replace the existing
+binary with the new one.
+
+**3.** Perform a rolling restart of Consul on your servers, leaving the leader agent
+for last. Only restart one server at a time. After restarting each server, validate
+that it has rejoined the cluster and is in sync with the leader by issuing the `consul info` command,
+and checking whether the `commit_index` and `last_log_index` fields have the same value.
+If done properly, this should avoid an unexpected leadership election due to loss of quorum.
+
+~> It is important to issue a `consul leave` command on each server node when shutting
```

Comment:
Recommending moving this detail to part of step #3. This makes sure someone doesn't do a leave as part of the rolling restart of Consul.

Perform a rolling restart of Consul on your servers, leaving the leader agent for last. This means running the `consul leave` command on your server agent, and then use a service management system (e.g., systemd, upstart, etc.) to restart the system. If you are not using a service management system, you must restart the agent manually. To validate that the agent has rejoined the cluster and is in sync with the leader, issuing the `consul info` command, and check whether the `commit_index` and `last_log_index` fields have the same value. If done properly, this should avoid an unexpected leadership election due to loss of quorum.

Reply:
I noticed that sometimes you have commands inline on the instructions, and some times the command is a stand alone. I made this recommendation where CLI commands are on their own lines as part of the step

Reply:
👍 I like that change. Incorporating it.